### PR TITLE
Improve consistency of field titles, labels, types across the schema

### PR DESCRIPTION
**Pull request by ahed-compucorp**
Overview
----------------------------------------
The purpose of this PR is to normalize field titles, labels & widgets across the schema to follow these conventions:

- The ID field should have `<html><type>Number</type></html>`
- For FK fields (which contain the ID of another entity), `<title>` should end with "ID" and `<html><label>` should not. E.g. `Address.contact_id` field should have `title` "Contact ID" and `label` "Contact".
- Titles and labels should *not* contain the name of its own entity. E.g. `Email.contact_id` should have the `title` "Contact ID" not "Email Contact ID" and the `label` should be "Contact" not "Email Contact".

### CODE REVIEW

**Review by seamuslee001 — COMMENTED** [review]
These seem to make sense but would note that `./setup.sh -g` would need to be run as well

**Comment by colemanw** [comment]
**Conflicting files**
`xml/schema/Campaign/CampaignGroup.xml`
`xml/schema/Campaign/Survey.xml`
`xml/schema/Contribute/ContributionPage.xml`
`xml/schema/Event/Participant.xml`

**Comment by eileenmcnaughton** [comment]
Argh - we should have merged quickly while it was unconflicted & passing - looks like it went stale!

@ahed-compucorp this is obviously a tricky PR to manage since it's inclined to go stale & run into test issues. Maybe if you drop out the last commit for now it will be passing again & we can move more quickly to merge

I'm happy to leave the DAO generation @seamuslee001 mentioned until all is merged

**Comment by colemanw** [comment]
@eileenmcnaughton MCs have been fixed but there are some test failures due to tests expecting certain field titles to be returned.
@ahed-compucorp is working on updating the tests.

**Review by colemanw — APPROVED** [review]
All these updates look good to me & will help the Search Kit UI feel consistent.

**Comment by ahed-compucorp** [comment]
@eileenmcnaughton I can regenerate DAO files with `./setup.sh -g` as @seamuslee001  mentioned 

I was feeling a bit overwhelmed by how DAO files are generated from theses xml files in `xml/schema`.

**Comment by colemanw** [comment]
This looks perfect. Thank you @ahed-compucorp

**Comment by eileenmcnaughton** [comment]
Great work @ahed-compucorp !

**Comment by mattwire** [comment]
@colemanw This PR has caused a regression https://github.com/civicrm/civicrm-core/pull/19531 - for example editing a paymentprocessor gives a fatal error "Unsupported html-element '' "

**Comment by colemanw** [comment]
@mattwire can you please try this fix: #19560

File: CRM/Core/DAO/Website.php [hunk 1]
```diff
@@ -124,19 +124,25 @@ public static function &fields() {
...
+          'title' => ts('Contact ID'),
           'description' => ts('FK to Contact ID'),
           'where' => 'civicrm_website.contact_id',
           'table_name' => 'civicrm_website',
           'entity' => 'Website',
           'bao' => 'CRM_Core_BAO_Website',
           'localizable' => 0,
           'FKClassName' => 'CRM_Contact_DAO_Contact',
+          'html' => [
+            'label' => ts("Contact"),
```
Comment (agileware-justin) [inline CRM/Core/DAO/Website.php:144]: Apologies in advise for commenting on a merged PR and spamming everyone.

But I was trying to track down a field label inconsistency in the Data Processor, https://lab.civicrm.org/extensions/dataprocessor/-/issues/90

Why is this field label "Contact" and not "Contact ID" ?

Comment (colemanw) [reply CRM/Core/DAO/Website.php:144]: The difference between "title" and "label" is that the title describes what the field contains (Contact ID) and the label is how a form label would appear (a Select field labeled "Contact").

Comment (agileware-justin) [reply CRM/Core/DAO/Website.php:144]: cool thanks @colemanw

**Pull request merged**